### ISSUE TITLE
DCC chat w/o requiring IRC server for initial handshake

### ISSUE THREAD

**Issue by Logicwax** [weechat/weechat#34]
mIRC and irssi already support this.   DCC chat on weechat works fine, but would be great to have to not rely on a 3rd party service at all, and go straight from IP to IP.

**Comment by flashcode** [comment]
Hi,

Sorry for the late reply.

Can you tell me more about such feature: DCC without IRC server? How does that work? Does that require a script that listens on some port, to get incoming DCC requests?

**Comment by Logicwax** [comment]
Wow, talk about resurrecting something I haven't thought about in 6 years!

Yes, that's how I believe mIRC did this (and irssi).  You could DCC connect directly by just having the other party IP address.

**Comment by flashcode** [comment]
Yes so if you give only IP, the remote should have something listening there.
So I suppose mIRC and irssi have some kind of plugin/feature which is a mini server listening on a port?

**Comment by Logicwax** [comment]
Yup, they listen on a socket.   That's why in mIRC you set a DCC listening port (regardless if its initiated via IRC server or direct)
